pkg/resources/configmaps.go

```diff
@@ -0,0 +1,325 @@
+        </storage>
+        <entry>
+          fields_strip_underscores true
+          fields_lowercase true
+        </entry>
+    </source>`
+var sourceConfigData3 = `
+    <source>
+        @type http
+        # Tag is not supported in yaml, must be set by request path (/icp-audit.http is required for validation and export)
+        port `
+var sourceConfigData4 = `
+        bind 0.0.0.0
+        body_size_limit 32m
+        keepalive_timeout 10s
+        <transport tls>
+          ca_path /fluentd/etc/https/ca.crt
+          cert_path /fluentd/etc/https/tls.crt
+          private_key_path /fluentd/etc/https/tls.key
+        </transport>
+        <parse>
+          @type json
+        </parse>
+    </source>
+    <filter icp-audit>
+        @type parser
+        format json
+        key_name message
+        reserve_data true
+    </filter>
+    <filter icp-audit.*>
+        @type record_transformer
+        enable_ruby true
+        <record>
+          tag ${tag}
+          message ${record.to_json}
+        </record>
+    </filter>`
+
+var splunkConfigData1 = `
+splunkHEC.conf: |-
+    <match icp-audit icp-audit.**>`
+var splunkDefaults = `
+        @type splunk_hec
+        hec_host SPLUNK_SERVER_HOSTNAME
+        hec_port SPLUNK_PORT
+        hec_token SPLUNK_HEC_TOKEN
+        ca_file /fluentd/etc/tls/splunkCA.pem
+        source ${tag}`
+var splunkConfigData2 = `
+    </match>`
+
+var qRadarConfigData1 = `
+remoteSyslog.conf: |-
+    <match icp-audit icp-audit.**>`
+var qRadarDefaults = `
+        @type copy
+        <store>
+            @type remote_syslog
+            host QRADAR_SERVER_HOSTNAME
+            port QRADAR_PORT_FOR_icp-audit
+            hostname QRADAR_LOG_SOURCE_IDENTIFIER_FOR_icp-audit
+            protocol tcp
+            tls true
+            ca_file /fluentd/etc/tls/qradar.crt
+            packet_size 4096
+            program fluentd
+            <format>
+                @type single_value
+                message_key message
+            </format>
+        </store>`
+var qRadarConfigData2 = `
+    </match>`
+
+// DataSplunk defines the struct for splunk-hec-config
+type DataSplunk struct {
+	Value string `yaml:"splunkHEC.conf"`
+}
+
+// DataQRadar defines the struct for remote-syslog-config
+type DataQRadar struct {
+	Value string `yaml:"remoteSyslog.conf"`
+}
+
+// BuildConfigMap returns a ConfigMap object
+func BuildConfigMap(instance *operatorv1alpha1.AuditLogging, name string) (*corev1.ConfigMap, error) {
+	reqLogger := log.WithValues("ConfigMap.Namespace", InstanceNamespace, "ConfigMap.Name", name)
+	metaLabels := LabelsForMetadata(FluentdName)
+	dataMap := make(map[string]string)
+	var err error
+	switch name {
+	case FluentdDaemonSetName + "-" + ConfigName:
+		dataMap[enableAuditLogForwardKey] = strconv.FormatBool(instance.Spec.Fluentd.EnableAuditLoggingForwarding)
+		type Data struct {
+			Value string `yaml:"fluent.conf"`
+		}
+		d := Data{}
+		err = yaml.Unmarshal([]byte(fluentdMainConfigData), &d)
```

Comment:
Should you be checking err, before setting the value?

Reply:
Unmarshal returns an error https://godoc.org/gopkg.in/yaml.v2#Unmarshal

Reply:
Let me ask a different way, If you get an err, at 185, should you test the err, before setting the value at 186?

Reply:
I see what you mean, I added breaks before the key, value assignment in the map if there is an error